docs/conf.py

```diff
@@ -110,6 +110,7 @@
     "secondary_sidebar_items": ["page-toc"],
     "pygment_light_style": "napari",
     "pygment_dark_style": "napari",
+    "announcement": "https://raw.githubusercontent.com/psobolewskiPhD/napari-docs/announcement/docs/_templates/custom-template.html",
```

Comment:
I feel like this should be a local path, not a link to a GH blob? e.g.

```suggestion
    "announcement": "_templates/custom-template.html",
```

? Does that work?

Also I would call the file something else since it's a full announcement, not a template?

The class name (*sidebar* message) is somewhat misleading too, but I guess we just inherit that...?

Reply:
It has to be accessible via http to make it work, so I think that won't work.
I think the name of the html file doesn't matter, but re-used everything from pydata-sphinx-theme **docs** as a first try.

Reply:
Ok I just found the docs for this (I might consider adding a link to the conf.py as a comment above the "announcement" key). Since you're not using any fancy html, I would suggest we mostly use the text directly?

https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/announcements.html#provide-local-html-in-your-theme

Alternatively, if we want to keep all old announcements around for reference, we could have a folder full of announcements, e.g. `https://napari.org/dev/_static/announcements/survey-2023.html`, and just point to those. (I guess then you want to first make a PR for the announcement html and then make one for adding it to the banner.)

Reply:
I'll definitely add comments to conf.py once we have it sorted.
Regarding just using text, my understanding is that will only add the announcement on the `dev` site, based on:
https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/announcements.html#insert-remote-html-with-javascript
and the conversation starting here:
https://github.com/pydata/pydata-sphinx-theme/pull/722#issuecomment-1153048144

So it has to be an http link to a file.

Do we need historic announcements in the deployed site? there will be history in our git repo right?
Also do we want to *always show* some sort of (generic) welcome announcement, ala the pydata-sphinx-theme docs? 
I'd vote no, because then real announcements stand out more...

So I think I'd vote to just having a fixed location for the announcement file and then comment/uncomment in conf.py when we want it to go live in a PR that edits the copy.

In this PR I'd just launch a generic annoucement so we can see it all works right before the survey, news, etc. to set up the machinery and then followup PRs with real announcements.
Can add a PR to the contributing docs re: how to make announcements.

Reply:
Wow — even reading it now it takes a while for me to parse that that is indeed the consequence of the html vs text announcement. Good job grokking this @psobolewskiPhD! 😅 

I agree — no default announcement.

But:

> So I think I'd vote to just having a fixed location for the announcement file and then comment/uncomment in conf.py when we want it to go live in a PR that edits the copy.

I think the opposite here: announcement URL is fixed and uncommented, and indeed that needs to be cherry-picked *urgently* for 0.4.19 so that it will pick up announcements. *Then* making or removing announcements amounts only to adding or removing an html file in a specific location. My proposal is something like `https://napari.org/dev/_static/announcements/announcement.html`. What do you think?

Reply:
I'm not sure I totally follow?
As far as I can tell what we need is:
1) `"announcement": <URL>` in our conf.py to make an announcement.
2) a accessible via http page with the announcement content.

So for 1 above it sounds like you're suggesting we always have the conf.py line pointing to that URL, but sometimes the file is not there? This will result in a 404 banner -- I already tested this by accident. To not have an announcement banner you need to delete/comment-out the `"announcement:"` part of conf.py
Maybe one could edit the file to exist but be empty? Let me try in this PR.

For 2 above the URL you suggest should work, fine by me. I'll change the path in this PR.

And yeah, I think we need the new theme and this PR cherry picked to 0.4.19 for this to work on napari.org (because it points to stable, which isn't rebuilt), although I can't actually figure out where/how the docs release happens.

Reply:
Wait, no, that URL won't work right now because that file isn't deployed so it's not there when the conf.py is used at docs build. chicken and egg problem. 
I suspect this is why pydata people just point to the GitHub source. 

So if we did want that, we'd need a PR that adds that file, merge and deploy it so it's there on napari.org, then PR back here to set/change the link in conf.py

Reply:
And again just to clarify (maybe?), my understanding is that with the http location approach you can **change** the announcement by changing the file without rebuilding docs--but they still need to be built with the conf.py flag enabled (and the proper theme version etc.)
So if we want this on stable soon, we need to push this plus the theme PR into 0.4.19 release docs.

Reply:
My understanding is the same as yours @psobolewskiPhD - I think that plan is good 👍🏻